frontend/alert/src/components/managealert/index.js

```diff
@@ -0,0 +1,189 @@
+        return 1;
+    });
+
+    return sortedAlerts.filter((alert) => {
+        if (filter.search.length > 0) {
+            return alert.section.includes(filter.search.toUpperCase());
+        }
+        return true;
+    });
+};
+
+const getActionPromise = (id, actionenum) => {
+    let body;
+    switch (actionenum) {
+        case AlertAction.Resubscribe:
+            body = { resubscribe: true };
+            break;
+        case AlertAction.Cancel:
+            body = { cancelled: true };
+            break;
+        case AlertAction.Delete:
+            body = { deleted: true };
+            break;
+        default:
+    }
+
+    return fetch(`/api/alert/api/registrations/${id}/`, {
+        method: "PUT",
+        credentials: "include",
+        mode: "same-origin",
+        headers: {
+            Accept: "application/json",
+            "Content-Type": "application/json",
+            "X-CSRFToken": getCsrf(),
+        },
+        body: JSON.stringify(body),
+    });
+};
+
+
+const actionHandler = callback => (id, actionenum) => {
+    getActionPromise(id, actionenum)
+        .then(res => callback());
+};
+
+// id_list is an array of ids
+const batchActionHandler = (callback, idList) => (actionenum) => {
+    Promise.all(idList.map(id => getActionPromise(id, actionenum)))
+        .then(res => callback());
+};
+
+const batchSelectHandler = (setAlertSel, currAlerts, alerts) => (checked) => {
+    const selMap = {};
+    alerts.forEach((alert) => {
+        selMap[alert.id] = false;
+    });
+    if (!checked) {
+        currAlerts.forEach((alert) => {
+            selMap[alert.id] = true;
+        });
+    }
+    setAlertSel(selMap);
+};
+
+const ManageAlertWrapper = () => {
+    // alerts processed directly from registrationhistory
+    const [alerts, setAlerts] = useState([]);
+    // state tracking the batch select button
+    const [batchSelected, setBatchSelected] = useState(false);
+    // state tracking which alert has been selected/ticked
+    const [alertSel, setAlertSel] = useState({});
+    // alerts after passing through frontend filters
+    const [currAlerts, setCurrAlerts] = useState([]);
+    const [numSelected, setNumSelected] = useState(0);
+    const [filter, setFilter] = useState({ search: "" });
+
+    useEffect(() => {
+        setCurrAlerts(filterAlerts(alerts, filter));
+    }, [alerts, filter, setCurrAlerts]);
+
+    useEffect(() => {
+        const selMap = {};
+        alerts.forEach((alert) => {
+            selMap[alert.id] = false;
+        });
+        setAlertSel(selMap);
+        setBatchSelected(false);
+    }, [alerts, setAlertSel]);
+
+    useEffect(() => {
+        setNumSelected(Object.values(alertSel).reduce((acc, x) => acc + x, 0));
+    }, [alertSel]);
+
+    useEffect(() => {
+        processAlerts(setAlerts);
+    }, []);
+
+    return (
+        <>
+            <ManageAlertHeader />
+            <ManageAlert
+                setFilter={setFilter}
+                numSel={numSelected}
```

Comment:
If this is just derived from `alertSel`, it might make sense to just have it computed in the `ManageAlert` component.